flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java

```diff
@@ -0,0 +1,235 @@
+		this.redisDataType = dataTypeDescription.getDataType();
+		this.additionalKey = dataTypeDescription.getAdditionalKey();
+
+	}
+
+	/**
+	 * Called when new data arrives to the sink, and forwards it to Redis channel.
+	 * <p> If redis data type is {@link RedisDataType#HASH} Redis HSET command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#LIST} Redis RPUSH command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#SET} Redis SADD command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#PUBSUB} Redis PUBLISH command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#STRING} Redis SET command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#HYPER_LOG_LOG} Redis PFADD command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#SORTED_SET} Redis ZADD command will be applied.
+	 *
+	 * @param input The incoming data
+	 */
+	@Override
+	public void invoke(IN input) throws Exception {
+		String key = redisSinkMapper.getKeyFromData(input);
+		String value = redisSinkMapper.getValueFromData(input);
+
+		switch (redisDataType) {
+			case HASH:
+				this.redisCommandsContainer.hset(this.additionalKey, key, value);
+				break;
+
+			case LIST:
+				this.redisCommandsContainer.rpush(key, value);
+				break;
+
+			case SET:
+				this.redisCommandsContainer.sadd(key, value);
+				break;
+
+			case PUBSUB:
+				this.redisCommandsContainer.publish(key, value);
+				break;
+
+			case STRING:
+				this.redisCommandsContainer.set(key, value);
+				break;
+
+			case HYPER_LOG_LOG:
+				this.redisCommandsContainer.pfadd(key, value);
+				break;
+
+			case SORTED_SET:
+				this.redisCommandsContainer.zadd(this.additionalKey, value, key);
+				break;
+
+			default:
+				throw new IllegalArgumentException("Cannot process such data type: " + redisDataType);
+		}
+
+	}
+
+	/**
+	 * Initializes the connection to Redis by either cluster or sentinels or single server
+	 *
+	 * @throws IllegalArgumentException if jedisPoolConfig, jedisClusterConfig and jedisSentinelConfig all are null
+     */
+	@Override
+	public void open(Configuration parameters) throws Exception {
+		if (jedisPoolConfig != null) {
+			this.redisCommandsContainer = RedisCommandsContainerBuilder.build(jedisPoolConfig);
+		} else if (jedisClusterConfig != null) {
+			this.redisCommandsContainer = RedisCommandsContainerBuilder.build(jedisClusterConfig);
+		} else if (jedisSentinelConfig != null) {
+			this.redisCommandsContainer = RedisCommandsContainerBuilder.build(jedisSentinelConfig);
+		} else {
+			throw new IllegalArgumentException("Jedis configuration not found");
+		}
+	}
+
+	/**
+	 * Closes commands container
+	 * @throws IOException if command container is unable to close.
+	 */
+	@Override
+	public void close() throws IOException {
+		if (redisCommandsContainer != null) {
+			try {
+				redisCommandsContainer.close();
+			} catch (IOException e) {
```

Comment:
Do we need to catch and log here? As exception is re-thrown there is no advantage in catch+log.